src/SerialPort/SerialPort.ts

```diff
@@ -86,11 +89,31 @@ export const createSerialPort = async (
         ipcRenderer.invoke(SERIALPORT_CHANNEL.SET, path, newOptions);
     };
 
-    const error = await ipcRenderer.invoke(
-        SERIALPORT_CHANNEL.OPEN,
-        options,
-        overwriteOptions
-    );
+    const tryAndOpen = async (retryCount: number) => {
+        try {
+            return (await ipcRenderer.invoke(
+                SERIALPORT_CHANNEL.OPEN,
+                options,
+                overwriteOptions
+            )) as string;
+        } catch (error) {
+            if (
+                (error as Error).message.includes(
+                    'PORT_IS_ALREADY_BEING_OPENED'
+                ) ||
+                retryCount === 0
+            ) {
+                return new Promise<string>(resolve => {
+                    setTimeout(async () => {
+                        resolve((await tryAndOpen(retryCount - 1)) as string);
+                    }, 50 + Math.random() * 100);
```

Comment:
Does it need to be random? This will wait from `50ms` and up to `149ms`, can't it simply wait `100ms`?